### PR TITLE
refactor: update interface to generate meta for single file

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

This pull request primarily refactors the `generate_tool_meta` function in the `src/promptflow/promptflow/_core/tool_meta_generator.py` file and updates its usages in other parts of the code. The refactoring removes `tool_dict` and `exception_dict` parameters from `generate_tool_meta` and makes it return a tuple of dictionaries instead. A new helper function `_generate_tool_meta_and_update_dict` is introduced to update the dictionaries. The usages of `generate_tool_meta` in other parts of the code are updated accordingly.

Refactoring of `generate_tool_meta`:

* [`src/promptflow/promptflow/_core/tool_meta_generator.py`](diffhunk://#diff-5cd3c0319063421c9ff9210ca997e3f001e5d1e2c9f97b912f1a3f233f9ee5a6L378-R393): The `generate_tool_meta` function no longer accepts `tool_dict` and `exception_dict` as parameters. Instead, it initializes these dictionaries internally and returns them as a tuple. Also, a docstring is added to this function to explain its purpose and usage.

Introduction of a new helper function:

* [`src/promptflow/promptflow/_core/tool_meta_generator.py`](diffhunk://#diff-5cd3c0319063421c9ff9210ca997e3f001e5d1e2c9f97b912f1a3f233f9ee5a6R406-R417): A new function `_generate_tool_meta_and_update_dict` is introduced. This function calls `generate_tool_meta`, gets the returned dictionaries, and updates the `tool_dict` and `exception_dict` passed to it.

Updating usages of `generate_tool_meta`:

* [`src/promptflow/promptflow/_core/tool_meta_generator.py`](diffhunk://#diff-5cd3c0319063421c9ff9210ca997e3f001e5d1e2c9f97b912f1a3f233f9ee5a6R429-R431): In the `generate_tool_meta_in_subprocess` function, the target of the multiprocessing process is changed from `generate_tool_meta` to `_generate_tool_meta_and_update_dict`.
* [`src/promptflow/promptflow/_sdk/_utils.py`](diffhunk://#diff-47208ac35b30920275fcd5e55d662647ef360129359bdc77fddd2a2157b6f47eL561-L562): In the `_generate_tool_meta` function, the initialization of `tool_dict` and `exception_dict` is removed, and the call to `generate_tool_meta` is updated to use the new return value. [[1]](diffhunk://#diff-47208ac35b30920275fcd5e55d662647ef360129359bdc77fddd2a2157b6f47eL561-L562) [[2]](diffhunk://#diff-47208ac35b30920275fcd5e55d662647ef360129359bdc77fddd2a2157b6f47eL571-R569)

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by PeiwenGaoMS — DISMISSED** [review]

**Review by zhengfeiwang — DISMISSED** [review]

**Review by zhengfeiwang — DISMISSED** [review]

File: src/promptflow/promptflow/_core/tool_meta_generator.py [hunk 1]
```diff
@@ -398,6 +412,19 @@ def generate_tool_meta(
...
+
+
+def _generate_tool_meta_and_update_dict(
+    working_dir: Path,
+    tools: Mapping[str, Mapping[str, str]],
+    tool_dict: dict,
+    exception_dict: dict,
+    prevent_terminate_signal_propagation: bool = False,
+):
+    _tool_dict, _exception_dict = generate_tool_meta(working_dir, tools)
```
Comment (PeiwenGaoMS) [inline src/promptflow/promptflow/_core/tool_meta_generator.py:425]: Pass `prevent_terminate_signal_propagation` to generate_tool_meta

File: src/promptflow/promptflow/_core/tool_meta_generator.py [hunk 2]
```diff
@@ -376,13 +376,27 @@ def generate_tool_meta_dict_by_file(path: str, tool_type: ToolType):
 def generate_tool_meta(
     working_dir: Path,
     tools: Mapping[str, Mapping[str, str]],
-    tool_dict: dict,
-    exception_dict: dict,
     prevent_terminate_signal_propagation: bool = False,
 ):
-    # This method might run in a child process, and when executed within a uvicorn app,
```
Comment (PeiwenGaoMS) [inline src/promptflow/promptflow/_core/tool_meta_generator.py:383]: Please keep this comments

Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/_core/tool_meta_generator.py:383]: Or move this logic to _generate_tool_meta_and_update_dict

File: src/promptflow/promptflow/_core/tool_meta_generator.py (hunk 1 above)
Comment (elliotzh) [reply src/promptflow/promptflow/_core/tool_meta_generator.py]: good catch, updated

File: src/promptflow/promptflow/_core/tool_meta_generator.py (hunk 2 above)
Comment (elliotzh) [reply src/promptflow/promptflow/_core/tool_meta_generator.py:383]: it has been moved to docstring

Comment (elliotzh) [reply src/promptflow/promptflow/_core/tool_meta_generator.py:383]: need more information to determine if we should move it to `_generate_tool_meta_and_update_dict`

**Review by zhengfeiwang — DISMISSED** [review]

**Review by PeiwenGaoMS — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**